lib/declare_schema/model/index_definition.rb

```diff
@@ -103,7 +100,7 @@ def equivalent?(rhs)
       end
 
       def with_name(new_name)
-        self.class.new(@model, @fields, table_name: @table_name, index_name: @index_name, unique: @unique, name: new_name)
+        self.class.new(@table_name, @columns, unique: @unique, name: new_name)
```

Comment:
I think we should also bring the `allow_equivalent` and `where` options along when using this method.

Reply:
Agreed. I'd already added `where` support and just now I added `allow_equivalent`. Although we've agreed on the need to redo the "index changes on column renames" ticket that introduced this method...and I don't expect this method will be needed after that.